test/Microsoft.AspNet.Mvc.TagHelpers.Test/TagHelperOutputExtensionsTest.cs

```diff
@@ -8,13 +8,258 @@
+                        new TagHelperAttributeList
+                        {
+                            { "first", "A" },
+                            { "second", "B" },
+                            { "third", "C" }
+                        },
+                        new[]
+                        {
+                            new TagHelperAttribute("first", "A"),
+                            new TagHelperAttribute("third", "C"),
+                            new TagHelperAttribute("dynamic", "value"),
+                            new TagHelperAttribute("secondDynamic", "another value"),
+                        }
+                    },
+                    {
+                        "first",
+                        new TagHelperAttributeList
+                        {
+                            { "third", "C" },
+                            { "dynamic", "value" },
+                            { "secondDynamic", "another value"}
+                        },
+                        new TagHelperAttributeList
+                        {
+                            { "first", "A" },
+                            { "second", "B" },
+                            { "third", "C" }
+                        },
+                        new[]
+                        {
+                            new TagHelperAttribute("first", "A"),
+                            new TagHelperAttribute("third", "C"),
+                            new TagHelperAttribute("dynamic", "value"),
+                            new TagHelperAttribute("secondDynamic", "another value"),
+                        }
+                    },
+                    {
+                        "first",
+                        new TagHelperAttributeList
+                        {
+                            { "third", "C" },
+                            { "dynamic", "value" },
+                            { "secondDynamic", "another value"}
+                        },
+                        new TagHelperAttributeList
+                        {
+                            { "first", "A" },
+                            { "secondDynamic", "another value"},
+                            { "second", "B" },
+                            { "third", "C" },
+                        },
+                        new[]
+                        {
+                            new TagHelperAttribute("third", "C"),
+                            new TagHelperAttribute("dynamic", "value"),
+                            new TagHelperAttribute("first", "A"),
+                            new TagHelperAttribute("secondDynamic", "another value"),
+                        }
+                    },
+                    {
+                        "first",
+                        new TagHelperAttributeList
+                        {
+                            { "third", "C" },
+                            { "dynamic", "value" },
+                            { "secondDynamic", "another value"}
```

Comment:
We modify attributes somewhat similarly. Same setup could occur: https://github.com/aspnet/Mvc/pull/2987/files#r37362253